src/app/Shared/Services/Api.service.tsx

```diff
@@ -1295,6 +1302,15 @@ export class ApiService {
         }`,
       { id: target.id! },
     ).pipe(
+      tap((resp) => {
+        if (resp.data == undefined) {
+          if (isGraphQLAuthError(resp)) {
```

Comment:
Would it make sense to extract this behaviour out from `getTargetMBeanMetrics` into the more general `this.graphql` function? This way the error handling applies to all GraphQL responses, not only MBean metrics responses.

Also, there is a `handleError` function in this class which would be nice to reuse, or to modify it to use these `isGraphQL(Auth/SSL)Error` functions if possible.

Reply:
Makes sense, I've reorganized it to throw an HttpError in this.graphql if a request comes back with errors and filter on the graphql exception in handleError.